worker-core/src/main/java/com/github/workerframework/core/WorkerCore.java

```diff
@@ -151,12 +151,11 @@ public CoreTaskCallback(final Codec codec, final WorkerStats stats, final Worker
          * Use the factory to get a new worker to handle the task, wrap this in a handler and hand it off to the thread pool.
          */
         @Override
-        public void registerNewTask(final TaskInformation taskInformation, final byte[] taskMessage, Map<String, Object> headers)
+        public void registerNewTask(final TaskInformation taskInformation, final TaskMessage taskMessage, Map<String, Object> headers)
             throws InvalidTaskException, TaskRejectedException
         {
             Objects.requireNonNull(taskInformation);
             stats.incrementTasksReceived();
-            stats.getInputSizes().update(taskMessage.length);
```

Comment:
Discussed and identified as unused, removed to save deserializing the taskmessage again.